src/main/groovy/lang/ObjectRange.java

```diff
@@ -472,37 +484,61 @@ public boolean hasNext() {
 
     @Override
     public List step(int step) {
+        return step((Number) step);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List step(Number step) {
         final IteratorClosureAdapter adapter = new IteratorClosureAdapter(this);
         step(step, adapter);
         return adapter.asList();
     }
 
     /**
-     * Increments by one
+     * Increments by step size
      *
      * @param value the value to increment
-     * @return the incremented value
+     * @param step  the value to increment by
+     * @return the incremented value or null, if there isn't any
      */
-    private static Object increment(Object value) {
+    protected static Comparable increment(Comparable value, Number step) {
         if (value instanceof Number) {
-            return plus((Number) value, 1);
-        } else {
-            return InvokerHelper.invokeMethod(value, "next", null);
+            return (Comparable) plus((Number) value, step);
+        }
+
+        for (int i = 0; compareLessThan(i, step); i++) {
+            final Comparable next = (Comparable) InvokerHelper.invokeMethod(value, "next", null);
+            if (!compareGreaterThan(next, value)) /* e.g. `next` of the last element */ {
```

Comment:
should work for enums also, where after the last element the first should be returned